src/framework/global/thirdparty/haw_logger/logger/log_base.h

```diff
@@ -24,19 +24,29 @@
 
 #define IF_LOGLEVEL(level)  if (haw::logger::Logger::instance()->isLevel(level))
 
-#define LOG_STREAM(type, tag, funcInfo) haw::logger::LogInput(type, tag, funcInfo).stream
-#define LOG(type, tag)  LOG_STREAM(type, tag, FUNCNAME(FUNC_INFO) + ": ")
-
-#define LOGE_T(tag) IF_LOGLEVEL(haw::logger::Normal) LOG(haw::logger::Logger::ERRR, tag)
-#define LOGW_T(tag) IF_LOGLEVEL(haw::logger::Normal) LOG(haw::logger::Logger::WARN, tag)
-#define LOGI_T(tag) IF_LOGLEVEL(haw::logger::Normal) LOG(haw::logger::Logger::INFO, tag)
-#define LOGD_T(tag) IF_LOGLEVEL(haw::logger::Debug) LOG(haw::logger::Logger::DEBG, tag)
-
-#define LOGE LOGE_T(LOG_TAG)
-#define LOGW LOGW_T(LOG_TAG)
-#define LOGI LOGI_T(LOG_TAG)
-#define LOGD LOGD_T(LOG_TAG)
-#define LOGN if (0) LOGD_T(LOG_TAG) // compiling, but no output
+#define LOG_STREAM(type, tag, funcInfo, color) haw::logger::LogInput(type, tag, funcInfo, color).stream
+#define LOG(type, tag, color)  LOG_STREAM(type, tag, FUNCNAME(FUNC_INFO) + ": ", color)
+
+#define LOGE_T(tag, color) IF_LOGLEVEL(haw::logger::Normal) LOG(haw::logger::Logger::ERRR, tag, color)
+#define LOGW_T(tag, color) IF_LOGLEVEL(haw::logger::Normal) LOG(haw::logger::Logger::WARN, tag, color)
+#define LOGI_T(tag, color) IF_LOGLEVEL(haw::logger::Normal) LOG(haw::logger::Logger::INFO, tag, color)
+#define LOGD_T(tag, color) IF_LOGLEVEL(haw::logger::Debug) LOG(haw::logger::Logger::DEBG, tag, color)
+
+#define LOGE LOGE_T(LOG_TAG, haw::logger::Red)
+#define LOGW LOGW_T(LOG_TAG, haw::logger::Yellow)
+#define LOGI LOGI_T(LOG_TAG, haw::logger::Green)
+#define LOGD LOGD_T(LOG_TAG, haw::logger::None)
+#define LOGN if (0) LOGD_T(LOG_TAG, haw::logger::None) // compiling, but no output
+
+#define LOGL LOGI_T(LOG_TAG, haw::logger::Black)
+#define LOGR LOGI_T(LOG_TAG, haw::logger::Red)
+#define LOGG LOGI_T(LOG_TAG, haw::logger::Green)
+#define LOGY LOGI_T(LOG_TAG, haw::logger::Yellow)
+#define LOGB LOGI_T(LOG_TAG, haw::logger::Blue)
+#define LOGM LOGI_T(LOG_TAG, haw::logger::Magenta)
+#define LOGP LOGI_T(LOG_TAG, haw::logger::Magenta)
+#define LOGC LOGI_T(LOG_TAG, haw::logger::Cyan)
+#define LOGH LOGI_T(LOG_TAG, haw::logger::White)
```

Comment:
I think this group should be removed.
If someone needs their own color in some place, then we can use `LOGI_T`